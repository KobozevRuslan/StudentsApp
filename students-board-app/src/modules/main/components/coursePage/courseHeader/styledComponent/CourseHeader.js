import styled from 'styled-components';

import { White, Nero } from '@assets/colors/colors';

export const HeaderMainWrapper = styled.div`
  background: ${Nero};
  .headerWrapper {
    max-width: 1240px;
    margin: 0 auto;
    font-size: 16px;
  }
`;

export const BlockWrapper = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: center;
  max-width: 1140px;
  .headerWrapperItem {
    display: flex;
    align-items: center;
    h2 {
      font-size: 32px;
      margin: 0 8px 0 11px;
    }
    .itemRatingCircle {
      background: ${White};
      border-radius: 50%;
      width: 36px;
      height: 36px;
      color: ${Nero};
      font-weight: bold;
      display: flex;
      align-items: center;
      justify-content: center;
      margin-left: 8px;
    }
  }
  .itemMentor {
    width: 165px;
    p {
      font-weight: bold;
    }
  }
`;

export const ItemDirection = styled.div`
  display: flex;
  flex-wrap: wrap;
  max-width: 850px;
  margin: 32px 0 24px;
  font-size: 20px;
  .itemDirectionName {
    background: ${White};
    color: ${Nero};
    border-radius: 6px;
    display: flex;
    align-items: center;
    padding: 12px;
    margin: 0 8px 8px 0;
  }
`;
