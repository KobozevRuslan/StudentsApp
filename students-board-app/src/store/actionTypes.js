export const CHANGE_LOADER = 'CHANGE_LOADER';
export const CHANGE_TAB = 'CHANGE_TAB';
export const AUTH_LOGIN = 'AUTH_LOGIN';
export const AUTH_IN = 'AUTH_IN';
