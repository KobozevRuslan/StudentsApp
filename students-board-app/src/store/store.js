import { applyMiddleware, createStore, combineReducers } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';
import createSagaMiddleware from 'redux-saga';

import authReducer from './reducers/authReducer';
import appReducer from './reducers/appReducer';
import rootSaga from './sagas/rootSaga';

const rootReducer = combineReducers({
  app: appReducer,
  auth: authReducer,
});

const logger = store => next => action => {
  console.log('action', action);
  let result = next(action);
  console.log('nex state', store.getState());
  return result;
};

const sagaMiddleware = createSagaMiddleware();

export const store = createStore(
  rootReducer,
  composeWithDevTools(applyMiddleware(sagaMiddleware, logger)),
);

sagaMiddleware.run(rootSaga);
