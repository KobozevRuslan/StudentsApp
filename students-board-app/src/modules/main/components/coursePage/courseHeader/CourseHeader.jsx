import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

import { PathMain } from '../../../../routePath';

import leftArrow from '@assets/image/LeftArrow.png';

import { BlockWrapper, HeaderMainWrapper, ItemDirection } from './styledComponent/CourseHeader';

const CourseHeader = ({ coursesPage }) => {
  const { name, rating, mentorName, technologies } = coursesPage;

  return (
    <HeaderMainWrapper>
      <div className={'headerWrapper'}>
        <BlockWrapper>
          <div className={'headerWrapperItem'}>
            <Link to={PathMain}>
              <img src={leftArrow} />
            </Link>
            <h2>{name}</h2>
            <div className={'itemRatingCircle'}>{rating}</div>
          </div>
          <div>
            <p>Short course description</p>
          </div>
        </BlockWrapper>
        <BlockWrapper>
          <ItemDirection>
            {technologies.map(tech => {
              return (
                <div className={'itemDirectionName'} key={tech}>
                  {tech}
                </div>
              );
            })}
          </ItemDirection>
          <div className={'itemMentor'}>
            <span>Mentor:</span>
            <p>{mentorName}</p>
          </div>
        </BlockWrapper>
      </div>
    </HeaderMainWrapper>
  );
};

CourseHeader.propTypes = {
  coursesPage: PropTypes.object.isRequired,
};

export default CourseHeader;
