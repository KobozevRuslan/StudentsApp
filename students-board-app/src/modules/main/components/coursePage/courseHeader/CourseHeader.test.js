import React from 'react';
import { Provider } from 'react-redux';

import CourseHeader from './CourseHeader';
import { store } from '../../../../../store/store';

describe('CourseHeader component', () => {
  const props = [
    {
      name: 'Front-end trainee course',
      rating: '4.6',
      technologies: ['Java Script', 'React', 'MobX-state-tree', 'CSS'],
      mentorName: 'Semeon Zaltz',
    },
  ];

  it('should render CourseHeader component with props', () => {
    const component = shallow(
      <Provider store={store}>
        <CourseHeader coursesPage={{ ...props }} />
      </Provider>,
    );
    expect(component).toMatchSnapshot();
  });
});
