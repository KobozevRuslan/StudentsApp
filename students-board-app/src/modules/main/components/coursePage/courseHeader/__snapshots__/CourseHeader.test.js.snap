// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CourseHeader component should render CourseHeader component with props 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "@@observable": [Function],
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "@@observable": [Function],
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <CourseHeader
    coursesPage={
      Object {
        "0": Object {
          "mentorName": "Semeon Zaltz",
          "name": "Front-end trainee course",
          "rating": "4.6",
          "technologies": Array [
            "Java Script",
            "React",
            "MobX-state-tree",
            "CSS",
          ],
        },
      }
    }
  />
</ContextProvider>
`;
